relocation.bs

```diff
@@ -796,10 +804,15 @@ ignored by overload resolution.
 
 #### Trivial relocation #### {#trivial-relocation}
 
-A class-type `T` is trivially relocatable if all the following is true:
+A relocating constructor of a class type `X` is trivial if it is not user-provided and
+if `X` has no virtual functions and no virtual base classes and if,
+for each direct base class and direct non-static data member of class type or array thereof,
+the relocation operation (which may be a relocating constructor or
+synthesized from copy/move constructor plus destructor) selected
+to relocate that base or member is trivial.
```

Comment:
Does that mean that a class that declares no relocation ctor is trivially relocatable?
What about a relocation ctor defined as deleted?

Reply:
Hmm. This is modeled on http://eel.is/c++draft/class.copy.ctor#11, so the "trivially relocatable" definition needs to be modeled on [class.prop].

Reply:
OK, so now:
* if it does not declare a relocation ctor, and one is not implicitly declared, then it does not have an *eligible* relocation ctor, but also it does not have a relocation ctor, so it is trivially relocatable iff it is trivially copyable.
* if its relocation ctor is deleted, that relocation ctor is not *eligible*, but it does have a relocation ctor, so it is not trivially relocatable even if it is trivially copyable.